src/components/CCPADialog.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import {
+  ModalDialog, Form, useToggle, ActionRow, Button,
+} from '@edx/paragon';
+import Cookies from 'js-cookie';
+
+const CCPA_COOKIE_NAME = 'edx_do_not_sell';
+
+const CCPADialog = ({ dialogIsOpen, closeCallback, baseURL }) => {
+  const [isOpen, open, close] = useToggle(dialogIsOpen);
+
+  const getDoNotSellCookie = () => Cookies.get(CCPA_COOKIE_NAME) === 'true';
+  const [personalizationChecked, setPersonalizationChecked] = useState(() => !getDoNotSellCookie());
+
+  useEffect(() => {
+    if (dialogIsOpen) {
+      setPersonalizationChecked(!getDoNotSellCookie());
+      open();
+    }
+  }, [dialogIsOpen, open]);
+
+  useEffect(() => {
+    if (!isOpen) {
+      closeCallback();
+    }
+  }, [closeCallback, isOpen]);
+
+  const setDoNotSellCookie = (value) => {
+    const { host } = new URL(baseURL);
+    // Use global domain (`.edx.org`) without the subdomain
+    const hostParts = host.split('.');
+    const domain = hostParts.length > 2
+      ? `.${hostParts.slice(-2).join('.')}`
+      : host;
+    const cookieOptions = host.startsWith('localhost')
+      ? {}
```

Comment:
nit: you might already be aware but it appears no cookie is set on `localhost` unless you opt to not pass `cookieOptions` on line 45 below, which made testing in the example app a bit unintuitive.

again, this is how it was rolled out for all the other backport versions, so I'd lean more towards the side of let's keep as is for now for consistency and maybe come back as address as tech debt later on.